### PR TITLE
Add support for custom command handler object

### PR DESCRIPTION
**Pull request by alexs1305**
Added the ability to implement and register a custom handler that will be able to execute logic before and after a command is executed.
solves #1 
solves #2

### CODE REVIEW

File: EasyCommand.AspNetCore.Tests/EasyCommandServiceCollectionExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public class EasyCommandServiceCollectionExtensionsTests
+    {
+        [Fact]
+        public void SuccessfullyRegistersCommand_NoOptions()
+        {
+            var services = CreateServiceProvider(
+                    t => t.AddEasyCommand());
+
+            var command = services.GetService<ExampleCommandWithRequestAndResult>();
```
Comment (creyke) [inline EasyCommand.AspNetCore.Tests/EasyCommandServiceCollectionExtensionsTests.cs]: Should this not use this.Command<T> for completeness? Assuming this.Command may, in time, mutate the command before returning it.

File: EasyCommand.AspNetCore/ControllerBaseExtensions.cs [hunk 2]
```diff
@@ -13,30 +15,118 @@ public static void RegisterScope(IServiceScope scope)
...
+
+            for (var i = handlerIndex; i >= 0; i--)
+            {
+                try
+                {
+                    await handlers[i].AfterExecutionAsync(command, result);
+                }
+                catch (Exception e)
+                {
+                    new Exception($"Unable to execute AfterExecutionAsync on {handlers[i].GetType()} {(hasAdditionalInformation ? $"after {additionalFailureInformation}" : null)}", e);
```
Comment (creyke) [inline EasyCommand.AspNetCore/ControllerBaseExtensions.cs]: Does this need to throw?

File: EasyCommand.AspNetCore.Tests/EasyCommandServiceCollectionExtensionsTests.cs (hunk 1 above)
Comment (alexs1305) [reply EasyCommand.AspNetCore.Tests/EasyCommandServiceCollectionExtensionsTests.cs]: in this instance we are just checking that a command is registered in the DI container, if we are to use the this.Command then we will be checking the ControllerBaseExtensions as well and we would also need to build the service as well to provide a controller.

File: EasyCommand.AspNetCore/ControllerBaseExtensions.cs (hunk 2 above)
Comment (alexs1305) [reply EasyCommand.AspNetCore/ControllerBaseExtensions.cs]: yes sorry will add this in

**Pull request merged**